guides/v2.3/graphql/caching.md

```diff
@@ -0,0 +1,101 @@
+---
+group: graphql
+title: GraphQL caching
+---
+
+Magento can cache pages rendered from the results of certain GraphQL queries with [full-page caching]({{page.baseurl}}/extension-dev-guide/cache/page-caching.html). Full-page caching improves response time and reduces the load on the server. Without caching, each page might need to run blocks of code and retrieve large amounts of information from the database. Only queries submitted with an HTTP GET operation can be cached. POST queries cannot be cached.
+
+## Cached queries
+
+The definitions for some queries include cache tags. Full page caching uses these tags to keep track of cached content. They also allow public content to be invalidated. Private content invalidation is handled on the client side.
+
+Magento caches the following queries:
+
+* `category`
+* `cmsBlocks`
+* `cmsPage`
+* `products`
+* `urlResolver`
+
+Magento explicitly disallows caching the following queries.
+
+* `cart`
+* `country`
+* `countries`
+* `currency`
+* `customAttributeMetadata`
+* `customer`
+* `customerDownloadableProducts`
+* `customerOrders`
+* `customerPaymentTokens`
+* `storeConfig`
+* `wishlist`
+
+[Define the GraphQL schema for a module]({{page.baseurl}}/graphql/develop/create-graphqls-file.html) describes the syntax of a valid query.
+
+## Caching with Varnish
+
+We recommend setting up Varnish as a reverse proxy to serve the full page cache in a production environment. See [Configure and use Varnish]({{page.baseurl}}/config-guide/varnish/config-varnish.html) for more information.
+
+Magento 2.3.2 updates the `vcl_hash` subroutine in the template `varnish.vcl` file and creates the `process_graphql_headers` subroutine for both Varnish 4.x and 5.x. To enable GraphQL caching, either generate a new template file, or edit the `default.vcl` file on your system to match the current default template.
+
+If you choose to edit an existing `default.vcl` file, update the `vcl_hash` subroutine to check whether the request URL contains `graphql`, as follows:
+
+```text
+sub vcl_hash {
+    if (req.http.cookie ~ "X-Magento-Vary=") {
+        hash_data(regsub(req.http.cookie, "^.*?X-Magento-Vary=([^;]+);*.*$", "\1"));
+    }
+
+    # For multi site configurations to not cache each other's content
+    if (req.http.host) {
+        hash_data(req.http.host);
+    } else {
+        hash_data(server.ip);
+    }
+
+    if (req.url ~ "graphql") {
```

Comment:
The condition should be if (req.url ~ "/graphql"), it was updated in the code.

Reply:
Fixed